app/models/split.rb

```diff
@@ -82,8 +119,34 @@ def create_decision!(params = {})
     build_decision(params).tap(&:save!)
   end
 
+  def registry
+    if try(:feature_incomplete?)
+      knock_out_weightings(super)
+    else
+      super
+    end
+  end
+
   private
 
+  def knock_out_weightings(registry_hash, to: "false")
+    target_variant = to.to_s
+    knock_out_weightings_if_possible(registry_hash, to: target_variant) || log_knockout_error(target_variant) && registry_hash
+  end
+
+  def knock_out_weightings_if_possible(registry_hash, to:)
+    found_key = false
+    knocked_out_registry = registry_hash.each_with_object({}) do |(k, _), h|
+      h[k] = (k.to_s == to ? (found_key = true && 100) : 0)
```

Comment:
So, I'm like 99% sure that `(found_key = true && 100)` will set `found_key` to `100`. Which, to be fair, still quacks the right way.

Reply:
Good eye. Yeah; I might just lean into that unless you have feels. The test coverage gets me comfortable with behavior and against regressions, but you tell me.

Reply:
If it were a more active codebase and I knew someone was going to have to come back through in the near term, I'd suggest expanding it into a full `if ... else` with an `h[k] =` assignment in each branch. As-is, it sort of feels like you're tucking away all of this knock-out logic into as small a corner of this class as it can fit. But as you said, as long as there are tests, maybe it's fine to let it be and see how far it can take us.

Reply:
rubocop will cut you if the last line of each fork assigns to the same var.